### DISCUSSION TITLE
Using air for testing

### DISCUSSION THREAD

**Discussion by skelouse** [air-verse/air#382] [Ideas]
[Here](https://github.com/taubyte/office-space/tree/main/commands/air) I implemented a simple way to test and hot-reload with air while also indicating testing build flags, a specific test, etc without having to create a config.  I'm inclined to improve it on there unless there is a want to add an `air test` command to this package. 

Essentially `air test <args...>` would take the arguments and dump them into a config as:
```toml
[build]
  cmd = "go test <args...>"
```

Current usage examples can be found in the [README](https://github.com/taubyte/office-space/blob/main/README.md): 
```shell
# Livereload a test using air, see: https://github.com/cosmtrek/air
# Note: be sure you have air installed `which air` => ~/go/bin/air

# Simple Live reload of all relative tests
$ asd air

# Matcher Live reload equivalent to `go test -v --run <some-test>
$ asd air <some-test>

# Live reload of all tests below a given directory equivalent to `go test -v ./...`
$ asd air ./...
```

**Comment by skelouse** [comment]
I just updated my air install and noticed there are arguments to update the config strings, but the use case below does not work as flags are parsed by air:

```shell
`air -build.cmd go test -v --run <TestToRun>`
```

**Comment by makiuchi-d** [comment]
[arelo](https://github.com/makiuchi-d/arelo) is the right choice for your usage:
```shell
arelo -p '**/*.go' -i '**/.*' -- go test -v --run <TestToRun>
```
